### ISSUE TITLE
lr_scheduler.step() doesn't run on every epoch on "epoch" mode.

### ISSUE THREAD

**Issue by daijian217** [PyTorchLightning/pytorch-lightning#7737]
Hi, there:
&ensp; I am using PyTorch lightning now. I appreciate you guys supporting a more elegant way to use PyTorch. 
&ensp; The question is that the lr_scheduler.step() is not called on every epoch even I set the interval as "epoch" and frequency as "1". The .step() will be called every 10 epochs if I set check_val_every_n_epoch to 10 for example.  By tracing the log and debug your source code, I found that .step() was only called after the validation step was called if I assigned the validation step and params. But I want to run .step() on every single epoch, not just the epoch that the validation step running.  Considering the running of validation step as a condition of running LR scheduler step() seems too strictly. 
&ensp; Can you guys support running the LR scheduler step() every epoch no matter what the validation interval is? Or there is already a satisfying way that I missed? The pytorch lightning version is 1.3.1.
Thanks.

**Comment by awaelchli** [comment]
Hi
 
I think this was recently updated here:
https://github.com/PyTorchLightning/pytorch-lightning/pull/7357

The lr scheduler now steps at the end of epoch unconditionally when interval is set to "epoch".

**Comment by daijian217** [comment]
HI
Glad to hear that. V1.4 seems can't be acquired through pip. I'll handle the issue manually, and expect the release.

**Comment by awaelchli** [comment]
Yes 1.4 is not released yet. You can install from master if you want and test if it works.

`pip install https://github.com/PyTorchLightning/pytorch-lightning/archive/master.zip`
